### ISSUE TITLE
[FeatureReq]: Dynamic content in the layout based on URL params for SSR

### ISSUE THREAD

**Issue by akaufmann** [sveltejs/sapper#322]
Let's say you wan't something like [Rich-Harris GH Profile](https://github.com/Rich-Harris) to be a SPA with dynamic nested routes. 
The structure would be (?):
```shell
routes/user/
⤷ _layout.html
    [username].html
    [username]
     ⤷   repositories.html
          stars.html
          followers.html
          ...
```
Because the user profile is visible on every page, it must be in the `_layout.html` file but we don't have access to the param `username`.

One solution is to get the `username` param from `document.location.pathname` but the `document` object is not available in the `preload` hook. This makes server side rendering not possible.

tl;dr: Is it possible to provide URL params to the `preload` hook in the layout?

**Comment by akaufmann** [comment]
Ah, it works when you put the layout in the `[username]` folder. I could have guessed earlier.

Thanks Rich for this great framework!
